comp/core/workloadmeta/collectors/internal/ecs/v4parser.go

```diff
@@ -81,18 +84,54 @@ func (c *collector) getTaskWithTagsFromV4Endpoint(ctx context.Context, task v1.T
 		return v1TaskToV4Task(task), errors.New(err)
 	}
 
-	taskWithTags, err := c.metaV3or4(metaURI, "v4").GetTaskWithTags(ctx)
-	if err != nil {
-		// If it's a timeout error, log it as debug to avoid spamming the logs as the data can be fetched in next run
-		if errors.Is(err, context.DeadlineExceeded) {
-			log.Debugf("timeout while getting task with tags from metadata v4 API: %s", err)
-		} else {
-			log.Warnf("failed to get task with tags from metadata v4 API: %s", err)
+	var returnTask v3or4.Task
+	// If we have tags, are collecting tags and have zero tags present in the cache
+	// Call getTaskWithTags
+	if c.hasResourceTags && c.collectResourceTags && len(rt.tags) == 0 && len(rt.containerInstanceTags) == 0 {
```

Comment:
With `c.collectResourceTags` , users now need to turn on two feature flags for ECS collection

- ecs_collect_resource_tags_ec2
- ecs_task_collection_enabled

Reply:
So I believe the two flags do something different. `ecs_task_collection_enabled` allows users to collect tasks and only if that is true does `ecs_collect_resource_tags_ec2`. I think this could allow us to turn `ecs_collect_resource_tags_ec2` on by default since it shouldn't matter if `ecs_task_collection_enabled` if false. What do you think?